sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/cpp/CppGenerator.java

```diff
@@ -1703,10 +1722,18 @@ private CharSequence generateFileHeader(
             "#if __cplusplus >= 201703L\n" +
             "#  include <string_view>\n" +
             "#  define SBE_NODISCARD [[nodiscard]]\n" +
+            "#  define SBE_USE_STRING_VIEW 1\n" +
             "#else\n" +
             "#  define SBE_NODISCARD\n" +
             "#endif\n\n" +
 
+            "#if __cplusplus >= 202002L\n" +
+            "#  if defined(SBE_ENABLE_SPAN)\n" +
+            "#    include <span>\n" +
+            "#    define SBE_USE_SPAN 1\n" +
+            "#  endif\n" +
+            "#endif\n\n" +
```

Comment:
Given that the `SBE_USE_STRING_VIEW` is defined if compiler has support for `string_view`. We should do the same for `SBE_USE_SPAN`, i.e. no need to have an additional `SBE_ENABLE_SPAN`.
The definition of `SBE_USE_SPAN` should be protected with `#ifndef`.